src/main/java/roomescape/controller/ReservationController.java

```diff
@@ -0,0 +1,69 @@
+package roomescape.controller;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import roomescape.exception.InvalidReservationException;
+import roomescape.exception.NotFoundReservationException;
+import roomescape.model.Reservation;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicLong;
+
+@Controller
+public class ReservationController {
+
+    private List<Reservation> reservations = new ArrayList<>();
+    private AtomicLong index = new AtomicLong(1);
+
+    // 예약 관리 페이지
+    @GetMapping("/reservation")
+    public String reservationPage() {
+        return "reservation";
+    }
+
+    // 예약 조회
+    @GetMapping("/reservations")
+    @ResponseBody
+    public List<Reservation> getReservations() {
+        return reservations;
+    }
+
+    // 예약 추가
+    @PostMapping("/reservations")
+    public ResponseEntity<Reservation> addReservation(@RequestBody Reservation newReservation) {
+        if (newReservation.getName() == null || newReservation.getName().isEmpty() ||
+                newReservation.getDate() == null || newReservation.getDate().isEmpty() ||
+                newReservation.getTime() == null || newReservation.getTime().isEmpty()) {
+            throw new InvalidReservationException("필요한 인자가 없습니다.");
+        }
+
+        long id = index.getAndIncrement();
+        newReservation.setId(id);
+        reservations.add(newReservation);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("Location", "/reservations/" + id);
+
+        return new ResponseEntity<>(newReservation, headers, HttpStatus.CREATED);
+    }
+
+    // 예약 삭제
+    @DeleteMapping("/reservations/{id}")
+    public ResponseEntity<Void> deleteReservation(@PathVariable Long id) {
+        Optional<Reservation> reservationToDelete = reservations.stream()
+                .filter(reservation -> reservation.getId().equals(id))
+                .findFirst();
```

Comment:
```suggestion
        Reservation reservationToDelete = reservations.stream()
                .filter(reservation -> reservation.getId().equals(id))
                .findFirst()
                .orElseThrow(() ->  new NotFoundReservationException("삭제할 예약이 없습니다."));
```

`Optional`을 최대한 사용 하지 않도록 수정해보세요!
`Optional`은 어떤 단점이 있을까요?

Reply:
검색해본 결과, Optional의 단점은

1. 메모리 할당 및 성능에 부정적인 영향
2. 직렬화 불가능
3. 값이 반드시 존재할 필요가 없다는 의미로 해석될 수 있어, API 설계가 일관되지 않을 수 있다.

라고 합니다.